hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/BinaryStreamImpl.java

```diff
@@ -18,7 +18,7 @@
  * @author Steve Ebersole
  */
 public final class BinaryStreamImpl extends ByteArrayInputStream implements BinaryStream {
-	private final int length;
+	private final long length;
```

Comment:
A java array can't have a `long` number of elements, so this change is unnecessary.